api/pkg/auth/service/service.go

```diff
@@ -92,57 +91,121 @@ func createChecksum(token string) string {
 	return hex.EncodeToString(hash[:])
 }
 
-func (r *request) insertData(ghUser goth.User, code string) error {
+func (r *request) insertData(gitUser goth.User, code, provider string) error {
+
+	var acc model.Account
+	var user model.User
+
+	userQuery := r.db.Model(&model.User{}).
+		Where("email = ?", gitUser.Email)
 
 	// Check if user exist
-	q := r.db.Model(&model.User{}).
-		Where("github_login = ?", ghUser.NickName)
+	err := userQuery.First(&user).Error
 
-	var user model.User
-	err := q.First(&user).Error
+	// If email doesn't exists in users table
 	if err != nil {
+
+		// Check whether username and provider are matching
+		accountQuery := r.db.Model(&model.Account{}).Where("username = ?", gitUser.NickName).Where("provider = ?", provider)
+		err = accountQuery.First(&acc).Error
+
 		// If user doesn't exist, create a new record
 		if err == gorm.ErrRecordNotFound {
+			acc.Name = gitUser.Name
+			acc.Username = gitUser.NickName
+			acc.AvatarURL = gitUser.AvatarURL
+			acc.Provider = provider
 
-			user.GithubName = ghUser.Name
-			user.GithubLogin = ghUser.NickName
-			user.Type = model.NormalUserType
-			user.AvatarURL = ghUser.AvatarURL
 			user.Code = code
+			user.Email = gitUser.Email
+			user.Type = model.NormalUserType
+
+			// User ID is set to zero so we need to update the value with (last inserted user_id+1)
+			lastUser := model.User{}
+			r.db.Model(&model.User{}).Last(&lastUser)
```

Comment:
Do we need to handle any error here ?

Reply:
it doesn't returns any error instead returns `*gorm.DB` which we don't need